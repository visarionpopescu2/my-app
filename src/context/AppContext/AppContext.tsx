import React from 'react';
import { APP_STATE } from "types";

const initialState: APP_STATE = {
  user: undefined,
  articles: [],
  setArticles: () => {}
}

const AppContext = React.createContext(initialState);

export default AppContext
