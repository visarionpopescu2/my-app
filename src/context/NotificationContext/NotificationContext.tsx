import React from 'react';
import { NOTIFICATION_TYPE } from 'types';

type NOTIFICATION = {
  openNotification: (message: string, type: NOTIFICATION_TYPE) => void
}

const initialState: NOTIFICATION = {
  openNotification: (message: string, type: NOTIFICATION_TYPE) => {}
}

const NotificationContext = React.createContext(initialState);

export default NotificationContext
