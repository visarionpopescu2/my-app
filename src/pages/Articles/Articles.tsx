import { useContext } from 'react';
import { Space, Typography } from 'antd';
import { ArticleCard } from 'components';
import { ARTICLE } from 'types';
import AppContext from 'context/AppContext/AppContext';
import NotificationContext from 'context/NotificationContext/NotificationContext';

function Articles() {
  const { articles, setArticles } = useContext(AppContext);
  const { openNotification } = useContext(NotificationContext);
  
  const removeArticleFromState = async (id: string) => {
    setArticles(articles.filter((item) => item.id+'' !== id));
  }
  
  const updateArticleInState = async (article: ARTICLE) => {
    setArticles(articles.map((item) => item.id === article.id ? article : item));
  }
  
  return (
    <Space direction="vertical" size="middle" style={{ display: 'flex' }}>
      <Typography.Title level={3} style={{marginTop: 0}}>All blog articles</Typography.Title>
      {articles?.map((article: ARTICLE) => (
        <ArticleCard
          key={`article-${article.id}`}
          article={article}
          onDeleteSuccess={removeArticleFromState}
          onEditSuccess={updateArticleInState}
        />
      ))}
    </Space>
  );
};

export default Articles;
