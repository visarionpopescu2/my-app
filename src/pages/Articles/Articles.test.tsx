import { render } from '@testing-library/react';
import Articles from './Articles';
import { MemoryRouter } from 'react-router-dom';
import AppContext from 'context/AppContext/AppContext';
import NotificationContext from '../../context/NotificationContext/NotificationContext';

const USER = {
  id: '1',
  name: 'name',
  username: 'username',
  email: 'email',
  phone: '123445',
  website: 'www.website.com',
  company: { name: 'company', bs: 'business', catchPhrase: 'description' },
  address: { street: 'street', suite: 'suite', zipcode: '1234', city: 'city', geo: { lat: '11', lng: '22' }}
};
const setArticles = jest.fn();
const openNotification = jest.fn();

describe('Articles', () => {
  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should render Articles component', () => {
    let view = render(
      <AppContext.Provider value={{ user: USER, articles: [], setArticles: setArticles }}>
        <NotificationContext.Provider value={{ openNotification: openNotification }}>
          <Articles />
        </NotificationContext.Provider>
      </AppContext.Provider>
    );
    expect(view.asFragment()).toMatchSnapshot();
  });
});
