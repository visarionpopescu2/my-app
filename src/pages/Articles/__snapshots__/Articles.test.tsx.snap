// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles should render Articles component 1`] = `
<DocumentFragment>
  <div
    class="ant-space css-dev-only-do-not-override-2i2tap ant-space-vertical ant-space-gap-row-middle ant-space-gap-col-middle"
    style="display: flex;"
  >
    <div
      class="ant-space-item"
    >
      <h3
        class="ant-typography css-dev-only-do-not-override-2i2tap"
        style="margin-top: 0px;"
      >
        All blog articles
      </h3>
    </div>
  </div>
</DocumentFragment>
`;
